File: app/src/main/java/ru/deltadelete/lab10/entities/Town.java
package ru.deltadelete.lab10.entities;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity(
        tableName = "towns",
        foreignKeys = {
                @ForeignKey(
                        entity = Country.class,
                        parentColumns = {"country_id"},
                        childColumns = {"country_id"},
                        onDelete = ForeignKey.CASCADE
                )
        }
)
public class Town {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "town_id")
    public int id;

    @ColumnInfo(name = "town_name")
    public String name;

    @ColumnInfo()
    public String description;

    @ColumnInfo(name = "country_id")
    public int countryId;
}

File: app/src/main/java/ru/deltadelete/lab10/database/entities/Town.kt
package ru.deltadelete.lab10.database.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.ForeignKey.Companion.CASCADE
import androidx.room.PrimaryKey

@Entity(
    tableName = "towns",
    foreignKeys = [ForeignKey(
        entity = Country::class,
        parentColumns = ["country_id"],
        childColumns = ["country_id"],
        onDelete = CASCADE
    )]
)
data class Town (
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "town_id")
    val id  : Int = 0,

    @ColumnInfo(name = "town_name")
    val name: String,

    @ColumnInfo
    val description: String,

    @ColumnInfo(name = "country_id", index = true)
    val countryId: Int
)